test/test_01_hw_03.py

```diff
@@ -0,0 +1,13 @@
+from unittest import mock
+
+from hw.hw_03_task_01 import cache
+
+
+def test_cache():
+    m = mock.Mock()
+    n = m
+    m = cache(times=2)(m)
+    for _ in range(4):
```

Comment:
It is better to do calls manually in that case.  
Also, you can do a couple of assers in the middle.